src/app.types.tsx

```diff
@@ -32,36 +32,37 @@ export interface CatalogueCategory {
   is_leaf: boolean;
   catalogue_item_properties?: CatalogueCategoryFormData[];
 }
+
 export interface AddManufacturer {
```

Comment:
You can also use what Joshua has here https://github.com/ral-facilities/inventory-management-system/blob/afcb642a56f54c0bd0a7238ec86e24355337ff92/src/app.types.tsx#L109 for edit, using `Partial` to effectively copy everything in the `CatalogueItemDetails` but allowing them to be undefined as well. This would also mean adding a separate Details that is extended by the others.